lib/ioc_container.dart

```diff
@@ -154,12 +150,14 @@ class IocContainerBuilder {
   void addSingleton<T>(
     T Function(
       IocContainer container,
-    ) factory,
-  ) =>
+    ) factory, {
+    void Function(T service)? dispose,
+  }) =>
       addServiceDefinition<T>(
         ServiceDefinition<T>(
           (container) => factory(container),
           isSingleton: true,
+          dispose: dispose,
```

Comment:
🤔

This looks correct. It seems like this was a actually a bug in the interface. 

Sorry for the slow response, but I will get around to merging this soon.

Thanks 🙏🏼